ci-operator/step-registry/ipi/conf/azure/sharednetwork/ipi-conf-azure-sharednetwork-commands.sh

```diff
@@ -5,20 +5,37 @@ set -o errexit
 set -o pipefail
 
 # TODO: move to image
-curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
+curl -L https://github.com/mikefarah/yq/releases/download/v4.22.1/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
 
 CONFIG="${SHARED_DIR}/install-config.yaml"
-PATCH=/tmp/install-config-sharednetwork.yaml.patch
 
-azure_region=$(/tmp/yq r "${CONFIG}" 'platform.azure.region')
+if [ -f "${SHARED_DIR}/customer_vnet_subnets.yaml" ]; then
+  vnet_name=$(cat ${SHARED_DIR}/customer_vnet_subnets.yaml | /tmp/yq '.platform.azure.virtualNetwork')
+  controlPlaneSubnet=$(cat ${SHARED_DIR}/customer_vnet_subnets.yaml | /tmp/yq '.platform.azure.controlPlaneSubnet')
+  computeSubnet=$(cat ${SHARED_DIR}/customer_vnet_subnets.yaml | /tmp/yq '.platform.azure.computeSubnet')
+  
+  PATCH="${SHARED_DIR}/install-config-qe-sharednetwork.yaml.patch"
 
-cat > "${PATCH}" << EOF
+  cat >> "${PATCH}" << EOF
+platform:
+  azure:
+    networkResourceGroupName: os4-common
+    virtualNetwork: ${vnet_name}
+    controlPlaneSubnet: ${controlPlaneSubnet}
+    computeSubnet: ${computeSubnet}
+EOF
+else
+  PATCH=/tmp/install-config-sharednetwork.yaml.patch
+
+  azure_region=$(/tmp/yq r "${CONFIG}" 'platform.azure.region')
+
+  cat > "${PATCH}" << EOF
 platform:
   azure:
     networkResourceGroupName: os4-common
     virtualNetwork: do-not-delete-shared-vnet-${azure_region}
     controlPlaneSubnet: subnet-1
     computeSubnet: subnet-2
 EOF
-
-/tmp/yq m -x -i "${CONFIG}" "${PATCH}"
+fi
+/tmp/yq ea -i "${CONFIG}" "${PATCH}"
```

Comment:
I run some local testing, seem like the command does not generate correct install-config:
```
[root@preserve-jialiu-ansible ~]# /tmp/yq --version
yq (https://github.com/mikefarah/yq/) version 4.22.1
[root@preserve-jialiu-ansible ~]# cat test.1 
platform:
  azure:
    virtualNetwork: 1.1
[root@preserve-jialiu-ansible ~]# cat test.2
platform:
  azure:
    networkResourceGroupName: os4-common
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2
[root@preserve-jialiu-ansible ~]# /tmp/yq ea -i test.1 test.2
[root@preserve-jialiu-ansible ~]# cat test.1
platform:
  azure:
    virtualNetwork: 1.1
[root@preserve-jialiu-ansible ~]# cat test.2
platform:
  azure:
    virtualNetwork: 1.1
---
platform:
  azure:
    networkResourceGroupName: os4-common
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2
```

Reply:
The 3.3.0 ym command works well.
```
[root@preserve-jialiu-ansible ~]# yq --version
yq version 3.3.0
[root@preserve-jialiu-ansible ~]# cat test.1 
platform:
  azure:
    virtualNetwork: 1.1
[root@preserve-jialiu-ansible ~]# cat test.2
platform:
  azure:
    networkResourceGroupName: os4-common
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2
[root@preserve-jialiu-ansible ~]# yq m -x -i test.1 test.2
[root@preserve-jialiu-ansible ~]# cat test.1
platform:
  azure:
    virtualNetwork: do-not-delete-shared-vnet-${azure_region}
    networkResourceGroupName: os4-common
    controlPlaneSubnet: subnet-1
    computeSubnet: subnet-2

```
Actually I prefer the 3.3.0 ym command, I can get most useful info via `--help` option, but I can not understand the help info from v4.22.1.

Reply:
The reason I used v4.22.1 is for getting the vnet_name, computeSubnet and controlPlane subnet from the YAML file. The documentation is not good for 3.3..0 yq command and they have changed the command structure after 4.18 so I thought of using 4.22 there. But yeah will look into how to use 3.3.0 yq to read yaml files